### PR TITLE
ssh/tailssh: don't swallow process exit code in be-child

### PR DESCRIPTION
**Pull request by bradfitz**
Thanks to @nshalman and @Soypete for debugging!

Updates #6054

### CODE REVIEW

File: ssh/tailssh/incubator.go [hunk 1]
```diff
@@ -283,7 +283,20 @@ func beIncubator(args []string) error {
...
-	return cmd.Run()
+	err = cmd.Run()
+	if ee, ok := err.(*exec.ExitError); ok {
+		ps := ee.ProcessState
+		code := ps.ExitCode()
+		if code < 0 {
+			// TODO(bradfitz): do we need to also check the syscall.WaitStatus
+			// and make our process look like it also died by signal/same signal
+			// as our child process? For now we just do the exit code.
+			fmt.Fprintf(os.Stderr, "[tailscale-ssh: process died: %v]\n", ps.String())
```
Comment (nshalman) [inline ssh/tailssh/incubator.go:294]: Should this be a more formal log line with `logf`? (I can't quite tell if you have access to that here...)

**Review by Soypete — APPROVED** [review]
LGTM! checking the ExitError should help debugging future issues as well

**Review by nshalman — APPROVED** [review]
Worked for my test case!

File: ssh/tailssh/incubator.go (hunk 1 above)
Comment (nshalman) [reply ssh/tailssh/incubator.go:294]: Looks like `logf := logger.Discard` so I guess not.

**Review by maisem — APPROVED** [review]

File: ssh/tailssh/incubator.go (hunk 1 above)
Comment (bradfitz) [reply ssh/tailssh/incubator.go:294]: logf at this point is a weirdo thing that writes via a high fd up to the child process, I believe.

This is definitely a weird environment.

**Pull request merged**